x/virtualgroup/keeper/keeper.go

```diff
@@ -481,6 +490,76 @@ func (k Keeper) StorageProviderExitable(ctx sdk.Context, spID uint32) error {
 	return nil
 }
 
+func (k Keeper) GetOrCreateGVGFamilyStatisticsWithinSP(ctx sdk.Context, spID uint32) *types.GVGFamilyStatisticsWithinSP {
+	store := ctx.KVStore(k.storeKey)
+
+	ret := &types.GVGFamilyStatisticsWithinSP{
+		StorageProviderId: spID,
+	}
+	bz := store.Get(types.GetGVGFamilyStatisticsWithinSPKey(spID))
+	if bz == nil {
+		return ret
+	}
+
+	k.cdc.MustUnmarshal(bz, ret)
+	return ret
+}
+
+func (k Keeper) DeleteGVGFamilyWithinSP(ctx sdk.Context, spID uint32, familyID uint32) {
+	gvgFamilyStatistics := k.MustGetGVGFamilyStatisticsWithinSP(ctx, spID)
+	index := -1
+	for i, id := range gvgFamilyStatistics.GlobalVirtualGroupFamilyIds {
+		if id == familyID {
+			index = i
+			break
+		}
+	}
+	if index != -1 {
```

Comment:
this can be put into the for loop, once meet the desired element, do the manipulation immediately.

Reply:
fixed